Decisions/scs-XXXX-v1-requirements-for-sso-identity-federation.md

```diff
@@ -0,0 +1,196 @@
+---
+title: Requirements for SSO identity federation
+type: Decision Record
+status: Draft
+track: IAM
+---
+
+# Introduction
+
+Our assumption is that there are use cases, where CSPs would like to be able to
+let customers access their SCS based services by identifying themselves with
+credentials that are stored and managed external to the CSP's SCS installation.
+
+This is based on the observation that prospective customers of a SCS based CSP
+sometimes already come equipped with an IAM solution of their choice, either on
+premises or e.g. as an external 3rd party cloud service. To ease onboarding of
+customer employees (or e.g. customer contracted 3rd party admin staff) as SCS
+users, it would be good to be able to consume these external identities in SCS.
+
+For customers this avoids the neccessity to explicitly maintain an additional
+dedicated account in SCS and this also reduces what SCS needs to do with
+respect to taking care of persisting user account information.
+
+To put it in other words, in SCS we would like to be able to delegate
+authentication to external identity providers and map those users to roles in
+SCS that can be used for authorization decisions when users access SCS services.
+
+In addition to user identities there we also see the necessity to support the
+use of "machine identites" (aka "workload identities" or "service accounts).
+These will probably be SCS-local accounts and have for example the purpose
+to grant CaaS workload access to storage resources served by the infrastructure
+layer. Exact architectural details for this are still in active discussion,
+but it is anticipated that the IdP component should be very useful in
+facilitating the integration.
+
+# Motivation for this document
+
+SCS has multiple service layers, like IaaS and CaaS, both of which running their
+own technological stack with specific internal models of accounts and
+authorization. OTOH there are also specific services like the "Status Page"
```

Comment:
I’d actually not reference the Status Page here, since that should run elsewhere and not rely on anything that is part of the infrastructure for which it provides “Status Infos” (otherwise if the castle is burnt down it can’t display “All things broken” :)